### PR TITLE
Druid dialect

### PR DESCRIPTION
**Pull request by pjain1**
- Adds DruidDialect
- SQL lib methods now accept dialect as parameter instead of same dialect being used throughout once toolbox is created

### CODE REVIEW

File: sql/src/main/java/com/rilldata/calcite/extensions/SqlCreateMetric.java [hunk 1]
```diff
@@ -108,7 +108,7 @@ private static Map<String, SqlNode> createMapFromList(SqlNodeList sqlNodes) thro
     for (SqlNode sqlNode : sqlNodes) {
       String name = sqlNode.accept(simpleIdentifierExtractor);
       if (name == null) {
-        throw new ParseException(String.format("Cannot find identifier for node %s", sqlNode));
+        throw new ParseException(String.format("Please provide an alias for %s", sqlNode));
       }
```
Comment (begelundmuller) [inline sql/src/main/java/com/rilldata/calcite/extensions/SqlCreateMetric.java:112]: Would it make sense to provide default aliases (like `measure1`, `measure2`, etc.) instead? (Note that I added this quickly, might be misinterpreting the context)

**Review by begelundmuller — APPROVED** [review]
I'm not seeing any test cases using the Druid dialect – am I missing something?

File: sql/src/main/java/com/rilldata/calcite/extensions/SqlCreateMetric.java (hunk 1 above)
Comment (pjain1) [reply sql/src/main/java/com/rilldata/calcite/extensions/SqlCreateMetric.java:112]: This is for using aliases for measures in the metrics view model. I think users should specify them explicitly as they will use these aliases in the sql queries in which they will use the defined model.

Comment (begelundmuller) [reply sql/src/main/java/com/rilldata/calcite/extensions/SqlCreateMetric.java:112]: I was thinking of the case where a user just writes the metrics view, and then a dashboard is generated from the metrics view. But your argument also makes sense. Let's stick with your way now and see if any users complain.

**Comment by pjain1** [comment]
> I'm not seeing any test cases using the Druid dialect – am I missing something?
@begelundmuller 
In sql/src/test/java/com/rilldata/CalciteTests.java, each of the test method now test for both Duckdb and Druid dialect. For testing operators [here](https://github.com/rilldata/rill-developer/pull/926/files#diff-5c7308e2070a0764f5eaa2a28dee7ebf3f127a777625c54a1e8c477f168ad1a3L282), expected operator output for both dialects are checked.

**Pull request merged**